### PR TITLE
Port commits to the right branch (http)

### PR DESCRIPTION
**Pull request by ktoso**
Ports https://github.com/akka/akka/pull/19100 to the right branch

### CODE REVIEW

**Comment by ktoso** [comment]
Will merge after validation.

**Comment by ktoso** [comment]
Yep... needs adjustments, I'll do those.

**Comment by ktoso** [comment]
FYI @ataraxer :)

File: akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala [hunk 1]
```diff
@@ -63,10 +63,15 @@ class DeflateCompressor extends Compressor {
     // `flush` and `finish` will return any size depending on the previous input.
     // This value will hopefully provide a good compromise between memory churn and
     // excessive fragmentation of ByteStrings.
-    new Array[Byte](size)
+    // We also make sure that buffer size stays within a reasonable range, to avoid
+    // draining deflator with too small buffer.
+    new Array[Byte](math.max(size, MinBufferSize))
+  }
```
Comment (2beaucoup) [inline akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala]: no need for these

Comment (ataraxer) [inline akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala]: This change actually fixes the empty string encoding case.

Comment (2beaucoup) [inline akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala]: These braces don't seem to fix anything. ;)

Comment (ataraxer) [inline akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala]: Ah, *these* :)
IMO dropping braces is OK for one-liners, but when there's a multi line comment directly above — it makes it a bit confusing.

Comment (ktoso) [inline akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala]: Yeah, technically not needed but I don't really mind them here ;-)

**Comment by ktoso** [comment]
Failure was unrelated (groupBy, which Roland resolved AFAIR).

**Pull request merged**